### ISSUE TITLE
arrow extensions

### ISSUE THREAD

**Issue by rickiesmooth** [tobilg/duckdb-nodejs-layer#4]
I'm trying to load the arrow extension on an ARM lambda, by executing the following statement:

```INSTALL arrow; LOAD arrow;```

and I'm met with the following error:

```bash
INFO	[Error: IO Error: Extension "/tmp/.duckdb/extensions/v0.8.0/linux_arm64/arrow.duckdb_extension"
could not be loaded: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /tmp/.duckdb/extensions/v0.8.0/linux_arm64/arrow.duckdb_extension)] 
{
  errno: -1, 
  code: 'DUCKDB_NODEJS_ERROR', 
  errorType: 'IO'
}
```

Do you maybe have some pointers on how to fix this?

**Comment by tobilg** [comment]
You can't install default extensions which weren't compiled on Amazon Linux 2... This leads to the exact error you're seeing.

**Comment by rickiesmooth** [comment]
gotcha so if i switch to x86 it should just work?

Is it cumbersome to compile the default extension for linux 2?

**Comment by tobilg** [comment]
No, this will be the same problem with x86 as well. This is due to GLIBC incompatibilities. I have a build pipeline for extensions here: https://github.com/tobilg/duckdb-nodejs-layer/blob/main/Dockerfile.spatial.x86_64

**Comment by tobilg** [comment]
Would you mind giving some insights about your use case @rickiesmooth?

**Comment by rickiesmooth** [comment]
I'm looking at `.github/workflows/main.yml`, and I have an idea on how to add some workflow steps for building something like `Dockerfile.arrow.arm64` using spatial as an example.

This would work great if I just need the arrow extension, but how would that work if I need both the spatial extension and the arrow extension? Have you considered having one "fat" image that holds all the official extensions which you can load, or would the layer size become too big? Alternatively, would it maybe be possible to build a layer with the correct GLIBC versions and [load the extensions directly from S3](https://duckdb.org/docs/extensions/working_with_extensions) ala `http://extensions.duckdb.org/v{release_version_number}/{platform_name}/{extension_name}.duckdb_extension.gz`? I'm pretty clueless on this subject, so sorry sounding uninformed. 😅 

My specific use case is using [arrowIPCStream](https://duckdb.org/docs/api/nodejs/reference#connectionarrowipcstreamsql-params-callback-) in combination with [AWS Lambda response streaming](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-response-streaming/)...so I only just need to install the arrow extension for that..

**Comment by tobilg** [comment]
You‘d have to build „all“ relevant extensions for Amazon Linux 2 and store them somewhere accessible via HTTPS. Which is a lot of effort honestly. It cost me about a dozen hours to get the spatial extension to work. Then, there’s the issue that some extensions need specific DuckDB versions, and are not compatible with others… It’s not a simple thing to accomplish…

**Comment by rickiesmooth** [comment]
hmm too bad, since the docs explicitly mention that downloading an extension from S3 could be helpful when building a lambda, I was hoping that that would make it a bit easier. 

What would you suggest? Make a similar layer as the spatial layer?

**Comment by tobilg** [comment]
Can you link the docs you mention? Well actually downloading from S3 is the easy part… Building and storing them is much more effort unfortunately.

**Comment by rickiesmooth** [comment]
I see, too bad! This was the doc I was reading: https://duckdb.org/docs/extensions/working_with_extensions

> Downloading an extension directly could be helpful when building a lambda or container that uses DuckDB. DuckDB extensions are stored in public S3 buckets, but the directory structure of those buckets is not searchable. As a result, a direct URL to the file must be used. To directly download an extension file, use the following format:
> 
> http://extensions.duckdb.org/v{release_version_number}/{platform_name}/{extension_name}.duckdb_extension.gz
> 
> For example:
> 
> http://extensions.duckdb.org/v0.8.1/windows_amd64/json.duckdb_extension.gz
> 
> The list of supported platforms may increase over time, but the current list of platforms includes:
> 
> linux_amd64_gcc4
> linux_amd64
> linux_arm64
> osx_amd64
> osx_arm64
> wasm_eh [DuckDB-Wasm’s extensions](https://duckdb.org/docs/api/wasm/extensions)
> wasm_mvp [DuckDB-Wasm’s extensions](https://duckdb.org/docs/api/wasm/extensions)
> windows_amd64
> windows_amd64_rtools
> See above for a list of extension names and how to pull the latest list of extensions.

e.g. https://extensions.duckdb.org/v0.8.1/linux_arm64/arrow.duckdb_extension.gz

**Comment by tobilg** [comment]
You can now try loading it from https://github.com/tobilg/duckdb-nodejs-layer/raw/main/release/extensions/v0.8.1/arrow.duckdb_extension Please give me some feedback whether this works or not. Thanks

**Comment by rickiesmooth** [comment]
Sorry but I'm unsure how to load it.

I naively tried:
```
INSTALL https://github.com/tobilg/duckdb-nodejs-layer/raw/main/release/extensions/v0.8.1/arrow.duckdb_extension; LOAD arrow;
```

**Comment by tobilg** [comment]
And that doesn't work? What's the error message? How did you test it? In Lambda directly? Did your allow unsigned extensions? https://duckdb.org/docs/archive/0.5.1/extensions/overview.html#unsigned-extensions

**Comment by rickiesmooth** [comment]
I get the following error when I test it in the Lambda directly:

```
Error: IO Error: Failed to read extension from "https://github.com/tobilg/duckdb-nodejs-layer/raw/main/release/extensions/v0.8.1/arrow.duckdb_extension": no such file] {
  errno: -1,
  code: 'DUCKDB_NODEJS_ERROR',
  errorType: 'IO'
}
```
my lambda:
```
// Instantiate DuckDB
const duckDB = new DuckDB.Database(":memory:", {
  allow_unsigned_extensions: "true",
});

// Create connection
const connection = duckDB.connect();

// Store initialization
let isInitialized = false;

export const handler = awslambda.streamifyResponse(async (event, responseStream, context) => {
  console.log(event, context);
  try {
    // Check if DuckDB has been initalized
    if (!isInitialized) {
      await query(`
        SET home_directory='/tmp';
        SET enable_http_metadata_cache=true;
        SET enable_object_cache=true;
        INSTALL 'https://github.com/tobilg/duckdb-nodejs-layer/raw/main/release/extensions/v0.8.1/arrow.duckdb_extension';
      `);
      isInitialized = true;
    }
```

While when I visit the link in my browser it downloads the extension.

**Comment by rickiesmooth** [comment]
ah I've tested it on arm, I can try with x86 later today!

**Comment by tobilg** [comment]
Haven't tried it, but I was pointed to https://github.com/duckdb/duckdb/pull/6049#issue-1563959687 in the DuckDB Discord.

**Comment by tobilg** [comment]
I think I was able to provide a solution for x86 based DuckDB Lambda functions:

Try to do the following:
* `SET custom_extension_repository = 'http://extensions.quacking.cloud';`
* `INSTALL arrow;`

Works with my DuckDB Lambda deployment. I haven't checked the functionality of the Arrow extension though.

**Comment by rickiesmooth** [comment]
wow that's slick! I've tried it out, but for some reason it tries to load an AMD extension?
```bash
Error: HTTP Error: Failed to download extension "arrow" at URL "http://extensions.quacking.cloud/v0.8.0/linux_amd64/arrow.duckdb_extension.gz
```

I've updated my lambda to use the x86 architecture so I'm unsure why it tries to download the `linux_amd64` architecture.

Full error:
```
2023-09-19T17:27:43.698Z	bbc0b6d7-ef1c-41fe-913d-738a332c6286	INFO	[Error: HTTP Error: Failed to download extension "arrow" at URL "http://extensions.quacking.cloud/v0.8.0/linux_amd64/arrow.duckdb_extension.gz"

Candidate extensions: "parquet"] {
  errno: -1,
  code: 'DUCKDB_NODEJS_ERROR',
  errorType: 'HTTP',
  statusCode: 403,
  response: '<?xml version="1.0" encoding="UTF-8"?>\n' +
    '<Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>45ZF0EY0Q54J8Z6A</RequestId><HostId>shYHxazrKW3GTSrD4UMUkXyV/tY8CoaFp/GnJgUHHgeI549VQHYT+yHHm5469Kj/RcL4jcZQK30=</HostId></Error>',
  reason: 'Forbidden',
  headers: {
    'Alt-Svc': 'h3=":443"; ma=86400',
    Connection: 'close',
    'Content-Type': 'application/xml',
    Date: 'Tue, 19 Sep 2023 17:27:43 GMT',
    Server: 'AmazonS3',
    'Transfer-Encoding': 'chunked',
    Via: '1.1 936f33bed45438343f0ef2adff442814.cloudfront.net (CloudFront)',
    'X-Amz-Cf-Id': 'F-TjTP5qVLObpyePoy5-D5kYDyZ6RymzXyhzxYfc3ULjsR0SQaPcOA==',
    'X-Amz-Cf-Pop': 'IAD89-C1',
    'X-Cache': 'Error from cloudfront'
  }
}
```

layer: <div id="cc" style="margin: 0px !important; padding: 0px !important;"><div id="c"><div id="app" class="awsui awsui-motion"><div data-analytics="root" data-analytics-type="tag" class=""><div class="awsui_root_lm6vo_1p5zj_127 awsui_root_1fj9k_1csvr_5" style="color: var(--color-text-body-default-ki9w9u,#16191f); display: flex; flex-direction: column; position: relative;"><div class="awsui_layout-wrapper_lm6vo_1p5zj_128" style="color: var(--color-text-body-default-ki9w9u,#16191f); display: flex; flex-direction: column; position: relative; min-height: calc(100vh - 102px);"><div class="awsui_layout_lm6vo_1p5zj_128" style="align-items: stretch; background-color: var(--color-background-layout-main-td853a,#f2f3f3); display: flex; flex: 1 1 auto;"><main class="awsui_layout-main_lm6vo_1p5zj_150" style="display: block; background-color: var(--color-background-layout-main-td853a,#f2f3f3); flex: 1 1 0%; min-width: 0px; position: relative;"><div><div class="awsui_content-wrapper_zycdx_1u7dy_97" style="padding-left: var(--space-scaled-2x-xxxl-tfmfie,40px); padding-right: var(--space-scaled-2x-xxxl-tfmfie,40px);"><div class="awsui_content-wrapper_lm6vo_1p5zj_174 awsui_content_1fj9k_1csvr_21" style="padding-bottom: var(--space-layout-content-bottom-ve4osc,20px); min-width: 280px;"><div data-analytics="refreshEnabledV2" data-analytics-type="variant"><div tabindex="-1" id="lambda-functionDesigner" class="lambda-page-container" data-analytics="functionDetail" data-analytics-type="eventContext" style="max-width: 1600px;"><div class="awsui_content_5gtk3_1pj4o_129"><div class="awsui_root_18582_frdpw_93 awsui_vertical_18582_frdpw_140 awsui_vertical-l_18582_frdpw_158" style="display: flex; flex-direction: column; row-gap: var(--space-scaled-l-08jb88, 20px);"><div class="awsui_child_18582_frdpw_97"><div class="code awsui_root_14rmt_16mbp_359 awsui_tabs_14rmt_16mbp_186" style="-webkit-font-smoothing: auto; border-collapse: initial; border-spacing: 0px; box-sizing: border-box; caption-side: top; color: var(--color-text-body-default-ki9w9u,#16191f); cursor: auto; direction: ltr; display: block; empty-cells: show; font-family: var(--font-family-base-4om3hr,&quot;Amazon Ember&quot;,&quot;Helvetica Neue&quot;,Roboto,Arial,sans-serif); font-size: var(--font-body-m-size-bf8hby,14px); font-stretch: normal; font-style: normal; font-variant: normal; font-weight: 400; hyphens: none; letter-spacing: normal; line-height: var(--font-body-m-line-height-wmg6vk,22px); list-style: outside none disc; tab-size: 8; text-align: left; text-align-last: auto; text-indent: 0px; text-shadow: none; text-transform: none; visibility: visible; white-space: normal; widows: 2; width: 1215px; word-spacing: normal;"><div class="awsui_tabs-content-wrapper_14rmt_16mbp_392 awsui_with-paddings_14rmt_16mbp_392"><div class="awsui_tabs-content_14rmt_16mbp_378 awsui_tabs-content-active_14rmt_16mbp_382" role="tabpanel" id="awsui-tabs-14-8919-code-panel" tabindex="0" aria-labelledby="awsui-tabs-14-8919-code" style="display: block; padding: var(--space-scaled-m-er48cl,16px) 0;"><div data-analytics="codeTab" data-analytics-type="eventContext"><div><div id="layersSectionContainer"><div data-analytics="layers" data-analytics-type="eventContext" class="awsui_root_wih1l_u3bvv_97 awsui_root_14iqq_14jts_97 awsui_variant-default_14iqq_14jts_119" data-selection-root="true" style="-webkit-font-smoothing: auto; overflow-wrap: break-word; border-collapse: initial; border-spacing: 0px; box-sizing: border-box; caption-side: top; color: var(--color-text-body-default-ki9w9u,#16191f); cursor: auto; direction: ltr; empty-cells: show; font-family: var(--font-family-base-4om3hr,&quot;Amazon Ember&quot;,&quot;Helvetica Neue&quot;,Roboto,Arial,sans-serif); font-size: var(--font-body-m-size-bf8hby,14px); font-stretch: normal; font-style: normal; font-variant: normal; font-weight: 400; hyphens: none; letter-spacing: normal; line-height: var(--font-body-m-line-height-wmg6vk,22px); list-style: outside none disc; position: relative; tab-size: 8; text-align: left; text-align-last: auto; text-indent: 0px; text-shadow: none; text-transform: none; visibility: visible; white-space: normal; widows: 2; word-spacing: normal; width: 1215px; background-color: var(--color-background-container-content-sgemmm,#fff); border-radius: var(--border-radius-container-jm6teg,0);"><div class="awsui_content-wrapper_14iqq_14jts_189" style="display: flex; flex-direction: column; width: 1215px;"><div class="awsui_content_14iqq_14jts_189" style="flex: 1 1 0%;"><div class="awsui_wrapper_wih1l_u3bvv_154 awsui_variant-container_wih1l_u3bvv_161 awsui_has-header_wih1l_u3bvv_168" role="region" tabindex="0" style="box-sizing: border-box; overflow-x: auto; position: relative; width: 1215px; padding-bottom: var(--space-table-content-bottom-r9l1c2,0); border-radius: var(--border-radius-container-jm6teg,0); box-shadow: 0 0 0 2px var(--color-border-item-focused-v8nq2m,#0073bb); outline: rgba(0, 0, 0, 0) dotted 2px; outline-offset: 2px;">

1 | duckdb-nodejs-x86 | 3 | nodejs14.x, nodejs16.x, nodejs18.x | - | arn:aws:lambda:us-east-1:041475135427:layer:duckdb-nodejs-x86:3
-- | -- | -- | -- | -- | --


</div><div class="awsui_sticky-scrollbar_faqt8_lhuk4_93 awsui_sticky-scrollbar-visible_faqt8_lhuk4_105 awsui_sticky-scrollbar-native-invisible_faqt8_lhuk4_108" style="bottom: 31px; display: block; height: 15px; overflow: auto hidden; position: sticky; width: 1215px; margin-top: -15px;"><div class="awsui_sticky-scrollbar-content_faqt8_lhuk4_102" style="height: 15px; width: 1350.66px;"></div></div></div></div></div></div></div></div></div></div></div></div></div></div></div></div></div></div></div></main><div class="awsui_drawer_1uo6m_vtgmd_102 awsui_drawer-closed_1uo6m_vtgmd_115 awsui_drawer-closed_1fj9k_1csvr_45" style="overflow-wrap: break-word; box-shadow: var(--shadow-panel-zt257m,0 1px 1px 0 #001c244d,1px 1px 1px 0 #001c2426,-1px 1px 1px 0 #001c2426); flex-shrink: 0; position: relative; min-width: 40px; z-index: 830;"><div class="awsui_drawer-content_1uo6m_vtgmd_122 awsui_drawer-content-clickable_1uo6m_vtgmd_137 awsui_tools_1fj9k_1csvr_33" style="background: var(--color-background-layout-panel-hover-58t7r5,#eaeded); overflow: auto; position: fixed; width: 40px; color: var(--color-text-interactive-default-dt26aj,#545b64); cursor: pointer; top: 71px; bottom: 31px;"><aside aria-label="Help panel" class="awsui_toggle_1uo6m_vtgmd_97" aria-hidden="false" style="box-sizing: border-box; padding: var(--space-xxs-9rrxti,4px) 7px;"><button class="awsui_tools-toggle_1fj9k_1csvr_41 awsui_toggle-button_16w0h_aq5wz_93" aria-label="Open help panel" type="button" aria-expanded="false" aria-haspopup="true" style="font-family: inherit; font-size: 14px; line-height: 1.15; margin: 0px; overflow: visible; text-transform: none; appearance: button; background: rgba(0, 0, 0, 0); border: 0px; color: currentcolor; cursor: pointer; padding: calc(var(--space-scaled-xxs-t7ij38, 4px) + 1px) calc(var(--space-xxs-9rrxti, 4px) + 1px);"><span class="awsui_icon_h11ix_zt9o9_98 awsui_size-normal-mapped-height_h11ix_zt9o9_151 awsui_size-normal_h11ix_zt9o9_147 awsui_variant-normal_h11ix_zt9o9_219" style="display: inline-block; position: relative; vertical-align: top; box-sizing: border-box; width: var(--size-icon-normal-wflv4k,16px); height: var(--font-body-m-line-height-wmg6vk,22px); padding: calc((var(--font-body-m-line-height-wmg6vk, 22px) - var(--size-icon-normal-wflv4k, 16px))/2) 0; color: currentcolor;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" focusable="false" aria-hidden="true"><circle class="stroke-linejoin-round" cx="8" cy="8" r="7"></circle><path d="M8 11V8H6"></path><path class="stroke-linejoin-round" d="M10 11H6"></path><path d="M7.99 5H8v.01h-.01z"></path></svg></span></button></aside></div></div></div></div></div></div></div></div></div><div id="f" role="contentinfo" ng-non-bindable="" style="margin: 0px; padding: 0px; border: 0px; width: 1375px; position: fixed; bottom: 0px; z-index: 1002;"><div id="footer"></div><div id="console-nav-footer"><div id="footer-content"></div><div id="bottom-drawer-container" data-testid="bottom-drawer-container"></div><div id="console-nav-footer-inner" data-testid="awsc-nav-footer" class="_awsc-footer__inner_swu42_45" aria-label="Footer" style="border-top: 1px solid rgb(84, 91, 100); color: rgb(255, 255, 255); font-family: &quot;Amazon Ember&quot;, &quot;Helvetica Neue&quot;, Roboto, Arial, sans-serif; font-size: 12px; background-color: rgb(35, 47, 62); line-height: 14px;"><div data-testid="awsc-nav-footer-content" id="awsc-nav-footer-content" class="_awsc-footer__inner__content_swu42_54" style="display: flex; margin: 0px; padding: 7px 16px; flex-wrap: wrap; justify-content: space-between; box-sizing: border-box;"><div class="_awsc-footer__inner__content__left_swu42_63" style="margin-right: 25px; display: inline-flex; align-items: center;"><div id="tools-experience-ingress-container"><button id="tools-experience-ingress" data-tools-experience-ingress-point="footer-btn" class="ingressButton-4-1-13" title="Open CloudShell" style="font-family: inherit; font-size: 12px; line-height: 1; margin: 0px 25px 0px 0px; overflow: visible; text-transform: none; appearance: button; color: rgb(255, 255, 255); border: none; cursor: pointer; display: flex; padding: 0px; background: transparent; align-items: center; font-weight: normal;"><svg xmlns="http://www.w3.org/2000/svg" version="2.0" aria-hidden="true" class="globalNav-0311" width="16" height="16" viewBox="0 0 16 16" style="color: rgb(255, 255, 255); font-family: &quot;Amazon Ember&quot;, &quot;Helvetica Neue&quot;, Roboto, Arial, sans-serif; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: center; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 47, 62); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><path d="M5 5l2.997 2.998L5 11m4.997-.002H12m3-7.626A2.374 2.374 0 0012.627 1H3.37A2.372 2.372 0 001 3.372v9.256a2.373 2.373 0 002.37 2.373h9.257A2.375 2.375 0 0015 12.628V3.372z" stroke="currentColor" stroke-width="2" stroke-miterlimit="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none"></path></svg><br class="Apple-interchange-newline">

runtime info:
*Runtime*: Node.js 18.x
*Architecture*: x86_64

**Comment by tobilg** [comment]
Yeah, that layer version is still on v0.8.0. I‘ll have to update it to 0.8.1 I guess. The error comes from the plug-in not being published for this version to S3. 

Can you try the spatial layer from https://github.com/tobilg/duckdb-nodejs-layer#arns? That should work…

**Comment by rickiesmooth** [comment]
soo close:
```
2023-09-19T19:33:03.199Z	5cdffb53-23b8-4b67-9cfd-b69f997aaee8	INFO	[
Error: Invalid Input Error: Extension "/tmp/.duckdb/extensions/a532702b9a/linux_amd64/arrow.duckdb_extension" version (v0.8.2-dev3190) does not match DuckDB version (0.8.2-dev2083)
] {
  errno: -1,
  code: 'DUCKDB_NODEJS_ERROR',
  errorType: 'Invalid Input'
}
```
I saw some comments in discord about dev versions, so if I understand correctly if we'll be able to use a stable version here everything (spatial & arrow) will just work?

**Comment by tobilg** [comment]
Hm, that’s somehow strange because I use the same layer version, and it worked in my case. 

But unfortunately that’s what I meant with the version conflicts in DuckDB’s Discord. It’s not easy to downgrade the versions because the extensions reference other commits in the submodules. Will have to think about how this can be solved.

**Comment by tobilg** [comment]
Can you do a `select version();`? It should show `0.8.2-dev2083` if you're using `duckdb-nodejs-spatial-x86:1... What puzzles me is that I use the above layer version, and it works flawlessly.

**Comment by rickiesmooth** [comment]
it does show `0.8.2-dev2083` when I do `select version();`

Forgot to mention that I needed to install and load it like so:

```
INSTALL arrow;
LOAD arrow;
```

Otherwise I'd get the following error:

```
Error: Catalog Error: Function with name "to_arrow_ipc" is not in the catalog, but it exists in the arrow extension.

To install and load the extension, run:
INSTALL arrow;
LOAD arrow;] {
  errno: -1,
  code: 'DUCKDB_NODEJS_ERROR',
  errorType: 'Catalog'
}
```

**Comment by tobilg** [comment]
So, I think I got it working finally... I published a new test layer which is built upon the latest supported [Spatial extension version](https://github.com/duckdblabs/duckdb_spatial) of [DuckDB](https://github.com/duckdb/duckdb/tree/9db510bd1105f883747baca317a9b63adedf0d8e). Then, I forked the Arrow extension and manually [updated to the same DuckDB commit](https://github.com/tobilg/arrow/tree/same-version).

I published this as a test Lambda layer: `arn:aws:lambda:us-east-1:041475135427:layer:duckdb-nodejs-spatial-test-x86:1`, and the GitHub Action Workflow to publish the regular layer is currently running. This will then be usable via `arn:aws:lambda:$REGION:041475135427:layer:duckdb-nodejs-spatial-x86:2`.

You should then be able to do:
* `LOAD '/opt/nodejs/node_modules/duckdb/extensions/spatial.duckdb_extension';`
* `INSTALL arrow;` (this loads the arrow extension from http://extensions.quacking.cloud/9db510bd11/linux_amd64/arrow.duckdb_extension.gz)
* `LOAD arrow;`

**Comment by rickiesmooth** [comment]
Awesome! I'll test it out tonight 🙌

**Comment by rickiesmooth** [comment]
this worked as a charm, thank you so much!

**Comment by tobilg** [comment]
Great! Curious about what you‘ll build though 😬

**Comment by rickiesmooth** [comment]
ping me on discord if you want a sneak peek 😉

**Comment by tobilg** [comment]
@rickiesmooth I just published the `spatial` and `arrow` extensions for DuckDB v0.9.0, which is available with `arn:aws:lambda:us-east-1:041475135427:layer:duckdb-nodejs-x86:5`:

* `SET custom_extension_repository = 'http://extensions.quacking.cloud';`
* `INSTALL arrow;`
* `LOAD arow;`

Works for me at least :-)

**Comment by rickiesmooth** [comment]
great! I'll give it a try and report back 🫡
